File: admin/src/types/friend.ts
// ========== 友链类型相关 ==========

// 友链类型实体
export interface FriendType {
  id: number
  name: string
  sort: number
  is_visible: boolean
  count: number // 该类型下的友链数量
}

// 友链类型列表响应数据
export interface FriendTypeListData {
  list: FriendType[]
  total: number
  page: number
  page_size: number
}

// 创建友链类型请求
export interface CreateFriendTypeRequest {
  name: string
  sort?: number
  is_visible?: boolean
}

// 更新友链类型请求
export interface UpdateFriendTypeRequest {
  name?: string
  sort?: number
  is_visible?: boolean
}

// ========== 友链相关 ==========

// 友链实体
export interface Friend {
  id: number
  name: string
  url: string
  description: string
  avatar: string
  screenshot: string // 网站截图
  sort: number // 排序值，范围1-10，默认5
  type_id: number | null // 友链类型ID
  type_name?: string // 友链类型名称
  is_invalid: boolean // 是否失效
  is_pending: boolean // 是否为待审核申请
  rss_url: string // RSS订阅地址
  rss_latime?: string // RSS订阅最后更新时间
  accessible: number // 可访问性状态: 0=正常, -1=忽略检查, >0=连续失败次数
}

// 友链列表查询参数
export interface FriendQuery {
  page?: number
  page_size?: number
}

// 友链列表响应数据
export interface FriendListData {
  list: Friend[]
  total: number
  page: number
  page_size: number
}

// 创建友链请求
export interface CreateFriendRequest {
  name: string
  url: string
  description?: string
  avatar?: string
  screenshot?: string
  sort?: number // 排序值，范围1-10，默认5
  type_id: number // 友链类型ID（必选）
  rss_url?: string // RSS订阅地址
}

// 更新友链请求
export interface UpdateFriendRequest {
  name?: string
  url?: string
  description?: string
  avatar?: string
  screenshot?: string
  sort?: number // 排序值，范围1-10
  type_id?: number | null
  is_invalid?: boolean // 是否失效
  is_pending?: boolean // 是否为待审核申请
  rss_url?: string // RSS订阅地址
  accessible?: number // 可访问性状态
}

File: server/internal/dto/friend.go
package dto

import "flec_blog/pkg/utils"

// ============ 友链类型 ============

// ListFriendTypeRequest 友链类型列表请求
type ListFriendTypeRequest struct {
	Page     int `form:"page" binding:"omitempty,min=1"`
	PageSize int `form:"page_size" binding:"omitempty,min=1,max=1000"`
}

// FriendTypeResponse 友链类型响应（前台用）
type FriendTypeResponse struct {
	ID   uint   `json:"id"`
	Name string `json:"name"`
	Sort int    `json:"sort"`
}

// CreateFriendTypeRequest 创建友链类型请求
type CreateFriendTypeRequest struct {
	Name      string `json:"name" binding:"required,max=50"`
	Sort      int    `json:"sort" binding:"omitempty"`
	IsVisible *bool  `json:"is_visible"`
}

// UpdateFriendTypeRequest 更新友链类型请求
type UpdateFriendTypeRequest struct {
	Name      string `json:"name" binding:"required,max=50"`
	Sort      int    `json:"sort" binding:"omitempty"`
	IsVisible *bool  `json:"is_visible"`
}

// FriendTypeListResponse 后台友链类型列表响应
type FriendTypeListResponse struct {
	ID        uint   `json:"id"`
	Name      string `json:"name"`
	Sort      int    `json:"sort"`
	IsVisible bool   `json:"is_visible"`
	Count     int    `json:"count"` // 该类型下的友链数量
}

// ============ 分组友链响应 ============

// FriendGroupResponse 友链分组响应
type FriendGroupResponse struct {
	TypeID   *uint                   `json:"type_id"`   // 类型ID，未分类为 null
	TypeName string                  `json:"type_name"` // 类型名称
	TypeSort int                     `json:"type_sort"` // 类型排序值
	Friends  []FriendInGroupResponse `json:"friends"`   // 该类型下的友链（已排序）
}

// FriendInGroupResponse 分组中的友链响应（精简版）
type FriendInGroupResponse struct {
	ID          uint   `json:"id"`
	Name        string `json:"name"`
	URL         string `json:"url"`
	Description string `json:"description"`
	Avatar      string `json:"avatar"`
	Screenshot  string `json:"screenshot"`
	Sort        int    `json:"sort"`
	IsInvalid   bool   `json:"is_invalid"` // 是否失效
}

// GroupedFriendsResponse 分组友链列表响应
type GroupedFriendsResponse struct {
	Groups       []FriendGroupResponse `json:"groups"`        // 友链分组列表
	TotalGroups  int                   `json:"total_groups"`  // 分组总数
	TotalFriends int                   `json:"total_friends"` // 友链总数
}

// ============ 通用友链请求 ============

// ListFriendRequest 友链列表请求
type ListFriendRequest struct {
	Page     int `form:"page" binding:"omitempty,min=1"`
	PageSize int `form:"page_size" binding:"omitempty,min=1,max=1000"`
}

// ============ 通用友链响应 ============

// FriendForWebResponse 前台友链响应
type FriendForWebResponse struct {
	ID          uint   `json:"id"`
	Name        string `json:"name"`
	URL         string `json:"url"`
	Description string `json:"description"`
	Avatar      string `json:"avatar"`
	Screenshot  string `json:"screenshot"`
	Sort        int    `json:"sort"`
	IsInvalid   bool   `json:"is_invalid"` // 是否失效
	TypeID      *uint  `json:"type_id"`    // 类型 ID
}

// ============ 后台友链管理请求 ============

// CreateFriendRequest 创建友链请求
type CreateFriendRequest struct {
	Name        string `json:"name" binding:"required,max=50"`
	URL         string `json:"url" binding:"required,url,max=255"`
	Description string `json:"description" binding:"max=500"`
	Avatar      string `json:"avatar" binding:"omitempty,url,max=255"`
	Screenshot  string `json:"screenshot" binding:"omitempty,url,max=255"`
	Sort        int    `json:"sort" binding:"omitempty,min=1,max=10"`
	TypeID      *uint  `json:"type_id" binding:"omitempty"`
	RSSUrl      string `json:"rss_url" binding:"omitempty,url,max=500"` // RSS订阅地址
}

// UpdateFriendRequest 更新友链请求
type UpdateFriendRequest struct {
	Name        string `json:"name" binding:"required,max=50"`
	URL         string `json:"url" binding:"required,url,max=255"`
	Description string `json:"description" binding:"max=500"`
	Avatar      string `json:"avatar" binding:"omitempty,url,max=255"`
	Screenshot  string `json:"screenshot" binding:"omitempty,url,max=255"`
	Sort        int    `json:"sort" binding:"omitempty,min=1,max=10"`
	IsInvalid   *bool  `json:"is_invalid"` // 是否失效
	IsPending   *bool  `json:"is_pending"` // 是否为待审核申请
	TypeID      *uint  `json:"type_id" binding:"omitempty"`
	RSSUrl      string `json:"rss_url" binding:"omitempty,url,max=500"` // RSS订阅地址
	Accessible  *int   `json:"accessible"`                              // 可访问性状态
}

// ============ 前台友链申请 ============

// ApplyFriendRequest 申请友链请求
type ApplyFriendRequest struct {
	Name        string `json:"name" binding:"required,max=50"`             // 网站名称
	URL         string `json:"url" binding:"required,url,max=255"`         // 网站链接
	Description string `json:"description" binding:"required,max=500"`     // 网站描述
	Avatar      string `json:"avatar" binding:"required,url,max=255"`      // 网站头像/logo
	Screenshot  string `json:"screenshot" binding:"omitempty,url,max=255"` // 网站截图（可选）
}

// ============ 后台友链管理响应 ============

// FriendListResponse 后台友链列表响应
type FriendListResponse struct {
	ID          uint            `json:"id"`
	Name        string          `json:"name"`
	URL         string          `json:"url"`
	Description string          `json:"description"`
	Avatar      string          `json:"avatar"`
	Screenshot  string          `json:"screenshot"`
	Sort        int             `json:"sort"`
	IsInvalid   bool            `json:"is_invalid"`
	IsPending   bool            `json:"is_pending"`
	TypeID      *uint           `json:"type_id"`
	TypeName    string          `json:"type_name,omitempty"`
	RSSUrl      string          `json:"rss_url"`
	RSSLatime   *utils.JSONTime `json:"rss_latime,omitempty"`
	Accessible  int             `json:"accessible"`
}
